### ISSUE TITLE
Docker image missing bcrypt

### ISSUE THREAD

**Issue by Diftraku** [pypiserver/pypiserver#224]
When running pypiserver with the official Docker image, uploading new packages fails due to missing bcrypt backend:

```
172.17.0.1 - - [09/Nov/2018 13:56:42] "POST / HTTP/1.0" 500 753
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/pypiserver-1.2.4-py3.6.egg/pypiserver/bottle.py", line 923, in _handle
    return route.call(**args)
  File "/usr/local/lib/python3.6/site-packages/pypiserver-1.2.4-py3.6.egg/pypiserver/bottle.py", line 1845, in wrapper
    rv = callback(*a, **ka)
  File "/usr/local/lib/python3.6/site-packages/pypiserver-1.2.4-py3.6.egg/pypiserver/_app.py", line 50, in protector
    if not config.auther(*request.auth):
  File "/usr/local/lib/python3.6/site-packages/pypiserver-1.2.4-py3.6.egg/pypiserver/core.py", line 90, in auth_by_htpasswd_file
    return htPsswdFile.check_password(username, password)
  File "/usr/local/lib/python3.6/site-packages/passlib/apache.py", line 852, in check_password
    ok, new_hash = self.context.verify_and_update(password, hash)
  File "/usr/local/lib/python3.6/site-packages/passlib/context.py", line 2428, in verify_and_update
    if not record.verify(secret, hash, **clean_kwds):
  File "/usr/local/lib/python3.6/site-packages/passlib/utils/handlers.py", line 761, in verify
    return consteq(self._calc_checksum(secret), chk)
  File "/usr/local/lib/python3.6/site-packages/passlib/handlers/bcrypt.py", line 530, in _calc_checksum
    self._stub_requires_backend()
  File "/usr/local/lib/python3.6/site-packages/passlib/utils/handlers.py", line 2221, in _stub_requires_backend
    cls.set_backend()
  File "/usr/local/lib/python3.6/site-packages/passlib/utils/handlers.py", line 2123, in set_backend
    return owner.set_backend(name, dryrun=dryrun)
  File "/usr/local/lib/python3.6/site-packages/passlib/utils/handlers.py", line 2143, in set_backend
    raise default_error
passlib.exc.MissingBackendError: bcrypt: no backends available -- recommend you install one (e.g. 'pip install bcrypt')
```

The container was created with the following options:
```bash
docker run -d --restart=always --name pypiserver -p 8080:8080 -v /mnt/pypi/packages:/data/packages -v /mnt/pypi/.htpasswd:/data/.htpasswd pypiserver/pypiserver:latest -P .htpasswd packages
```

**Comment by mplanchard** [comment]
Good catch! I'll have an update out shortly

**Comment by mplanchard** [comment]
@Diftraku could you please try again using `pypiserver/pypiserver:unstable`?

**Comment by Diftraku** [comment]
Uploading works without a hitch again, although I had to exec into the container to take ownership of the `packages` directory before pypiserver was able to write the new package to disk.

This may have something to do with how the mounts are setup and using unstable release but is most likely unrelated to this issue.

**Comment by mplanchard** [comment]
Awesome! I was able to upload locally without changing the ownership, but I'll poke around for a bit to make sure that's not a general problem.

**Comment by mplanchard** [comment]
Updated docker image should now be available at `pypiserver/pypiserver:latest` and `pypiserver/pypiserver:v1.2.5`. I included a minor fix to the pypiserver group in #228 in that image as well, so hopefully everything will now be working as expected.

**Comment by Diftraku** [comment]
Tested this just now and everything works. I restored the files on disk to be owned by root and uploads work as expected.

**Comment by kellycampbell** [comment]
I'm still seeing this issue. I looked for the bcrypt files in /usr/lib/python3.6/site-packages, but that didn't exist. I also looked under /usr/local/lib/python3.6/site-packages.

When I tried building my own docker image, it looks like apk isn't adding any files, so I researched and tried `apk add py3-bcrypt`. This instlled, but pulled in the default apk for python3.

Then I added the /usr/lib location to PYTHONPATH, but now theres other missing dependecies, six and cffi.

 I will try to get a working docker build

**Comment by mplanchard** [comment]
This should be resolved by #239, which has just been merged into master. The `unstable` image has already built, and I'll be cutting a `v1.2.7` release here in the next few minutes. Please feel free to re-open this issue if the problem is not resolved!
